site/src/components/layout.md

```diff
@@ -0,0 +1,152 @@
+| ------------------- | ------------------------- |
+| Narrow - Widescreen | All elements are stacked. |
+
+### Holy Grail layout
+
+Named after a layout that [historically was difficult to implement](<https://en.wikipedia.org/wiki/Holy_grail_(web_design)>) the holy grail is stacked on narrow widths and progressively shows up to three columns as the screen widens.
+
+How do I apply it? `layout-holy-grail` to the `layout` element.
+Required elements: all, except `layout-body-hero`.
+Allowed elements: all.
+
+The following block is arranged from narrow widths on the left to wider widths on the right.
+
+```Text
+  ┌────────────┐ ┌──────────────┐ ┌──────────────────────┐
+  │ Header     │ │Header        │ │ Header               │
+  ├────────────┤ ├──────────────┤ ├──────────────────────┤
+  │ Hero       │ │Hero          │ │ Hero                 │
+  ├────────────┤ ├─────┬────────┤ ├─────┬────────┬───────┤
+  │ Menu       │ │Menu │ Main   │ │Menu │ Main   │ Aside │
+  ├────────────┤ │     │        │ │     │        │       │
+  │ Main       │ │     │        │ │     │        │       │
+  ├────────────┤ │     ├────────┤ │     │        │       │
+  │ Aside      │ │     │ Aside  │ │     │        │       │
+  ├────────────┤ ├─────┴────────┤ ├─────┴────────┴───────┤
+  │ Footer     │ │Footer        │ │ Footer               │
+  └────────────┘ └──────────────┘ └──────────────────────┘
+```
+
+The specification for this layout is as follows.
+
+| Screensize | Behavior                                                                                                                      |
+| ---------- | ----------------------------------------------------------------------------------------------------------------------------- |
+| Narrow     | All elements are stacked.                                                                                                     |
+| Tablet     | Menu and main are side by side. Main is wider than menu. Aside has a collapsed height and is tucked under main.               |
+| Desktop    | Menu, main, aside are side by side. Main is wider than menu.                                                                  |
+| Widescreen | Same as desktop with scaling gutter that keeps combined width of menu, main, aside to the width of the widescreen breakpoint. |
+
+## Accessibility and ARIA landmarks
+
+Due to the [recommendation that all content be rendering into ARIA landmark regions](https://dequeuniversity.com/rules/axe/4.7/region), the hero element should be a `<section>` element with an unique aria-label (or aria-labelled by).
+
+## Advanced topic - switching layouts on the fly
+
+Because layouts generally contain the same elements and because the current layout is determined by a singular class on the `.layout` element, changing layout is as easy as swapping out a class.
+
+```JavaScript
+document.documentElement.classList.remove('layout-single');
+document.documentElement.classList.remove('layout-holy-grail');
```

Comment:
If this example is meant to demonstrate a swap from single to holy grail, should the second line be adding `layout-holy-grail` class instead?

Reply:
Good catch! Updated!